runtime/src/main/java/org/corfudb/runtime/collections/Table.java

```diff
@@ -0,0 +1,217 @@
+     * @param fullyQualifiedTableName Fully qualified table name.
+     * @param valueSchema             Value schema to identify secondary keys.
+     * @param corfuRuntime            Connected instance of the Corfu Runtime.
+     * @param serializer              Protobuf Serializer.
+     */
+    @Nonnull
+    public Table(@Nonnull final String namespace,
+                 @Nonnull final String fullyQualifiedTableName,
+                 @Nonnull final V valueSchema,
+                 @Nullable final M metadataSchema,
+                 @Nonnull final CorfuRuntime corfuRuntime,
+                 @Nonnull final ISerializer serializer) {
+
+        this.corfuRuntime = corfuRuntime;
+        this.namespace = namespace;
+        this.fullyQualifiedTableName = fullyQualifiedTableName;
+        if (metadataSchema != null) {
+            this.metadataOptions = MetadataOptions.builder()
+                    .metadataEnabled(true)
+                    .defaultMetadataInstance(metadataSchema)
+                    .build();
+        } else {
+            this.metadataOptions = MetadataOptions.<M>builder().build();
+        }
+        this.corfuTable = corfuRuntime.getObjectsView().build()
+                .setTypeToken(new TypeToken<CorfuTable<K, CorfuRecord<V, M>>>() {
+                })
+                .setStreamName(this.fullyQualifiedTableName)
+                .setSerializer(serializer)
+                .setArguments(new ProtobufIndexer(valueSchema))
+                .open();
+    }
+
+    /**
+     * Begins a write after write transaction.
+     */
+    private boolean TxBegin() {
+        if (!TransactionalContext.isInTransaction()) {
+            corfuRuntime.getObjectsView()
+                    .TXBuild()
+                    .type(TransactionType.OPTIMISTIC)
+                    .build().begin();
+            return true;
+        }
+        return false;
+    }
+
+    /**
+     * Ends an ongoing transaction.
+     */
+    private void TxEnd() {
+        corfuRuntime.getObjectsView().TXEnd();
+    }
+
+    /**
+     * Create a new record.
+     *
+     * @param key      Key
+     * @param value    Value
+     * @param metadata Record metadata.
+     * @return Previously stored record if any.
+     */
+    @Nullable
+    CorfuRecord<V, M> create(@Nonnull final K key,
+                             @Nullable final V value,
+                             @Nullable final M metadata) {
+        boolean beganNewTxn = false;
+        try {
+            beganNewTxn = TxBegin();
+            return corfuTable.put(key, new CorfuRecord<>(value, metadata));
+        } finally {
+            if (beganNewTxn) {
+                TxEnd();
+            }
+        }
+    }
+
+    /**
+     * Fetch the value for a key.
+     *
+     * @param key Key.
+     * @return Corfu Record for key.
+     */
+    @Nullable
+    public CorfuRecord<V, M> get(@Nonnull final K key) {
+        return corfuTable.get(key);
```

Comment:
check if corfutable returns null

Reply:
Not needed. `@Nullable` clarifies to the user to expect null values.